include/boost/json/basic_parser.hpp

```diff
@@ -1094,30 +1100,33 @@ parse_string(
             ++cs;
             continue;
         }
-        if(BOOST_JSON_LIKELY(more_))
+
+        // flush if the stream is empty
```

Comment:
LOL the comment is not only unhelpful, but wrong. It should say "buffer" not "stream", and it should say "not empty."

Reply:
Hah, this proves exactly why this comment is needed. If you look at the `if` statement above, if tests `*cs`, meaning that this statement is only executed when the stream is empty :)